File: 卡码/代码随想录/C++/贪心/03-困难/LeetCode_0435.cpp
//435. 无重叠区间
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
	int eraseOverlapIntervals(vector<vector<int>>& intervals) {
		if(intervals.size()==0) return 0;
		sort(intervals.begin(),intervals.end(),[](vector<int>& a,vector<int>& b){
			return a[0]<b[0];
		});
		int res=0;
		for(int i=1;i<intervals.size();i++){
			if(intervals[i][0]<intervals[i-1][1]){
				res++;
				intervals[i][1]=min(intervals[i][1],intervals[i-1][1]);
			}
		}
		return res;
	}
};

int main(){
  system("pause");
  return 0;
}

File: 卡码/代码随想录/1/08-贪心算法/Java/greedy/src/main/java/com/kurumi/greedy/_03_困难/LeetCode_0435.java
//435. 无重叠区间
package com.kurumi.greedy._03_困难;

import java.util.Arrays;

public class LeetCode_0435 {

}
class Solution {
  public int eraseOverlapIntervals(int[][] intervals) {
    if(intervals.length==0) return 0;
    Arrays.sort(intervals,(a,b)->Integer.compare(a[0], b[0]));
    int res=0;
    for(int i=1;i<intervals.length;i++){
      if(intervals[i][0]<intervals[i-1][1]){
        res++;
        intervals[i][1]=Integer.min(intervals[i][1], intervals[i-1][1]);
      }
    }
    return res;
  }
}
